### PR TITLE
specify content-type for headers for POST

### PR DESCRIPTION
**Pull request by janetliuboston**
For POST and GET requests, the header always contains the media type of JSON for all requests: 'content-type: json/application'. However, some GET and POST requests do not send any media resources with the request. A web server taking requests that get media resources when they are not part of the API could fail with HTTP 415, Unsupported Media Type.

This PR creates methods that separate PostJson and PostEmpty, GetJson and GetEmpty, so the content types can be explicitly called.

### CODE REVIEW

**Comment by janetliuboston** [comment]
May also need to add 'Accept: <content-type>' to the header for completeness.

File: src/harness/request_handler.py [hunk 1]
```diff
@@ -75,15 +75,31 @@ def WithClientCertificate(self, client_cert, client_key):
...
 
 
-def _Request(url, request, config, is_post_method):
+def RequestGetEmpty(url, config):
+  return _Request(url, None, config, False, None)
+
+
+def RequestGetJson(url, request, config):
+  return _Request(url, request, config, False, 'application/json')
+
```
[the remark this answers is not in the document]
Comment (janetliuboston) [reply src/harness/request_handler.py]: Thank you. I have removed this and RequestGetEmpty, just use RequestGet.

File: src/harness/request_handler.py [hunk 2]
```diff
@@ -75,15 +75,27 @@ def WithClientCertificate(self, client_cert, client_key):
...
+def RequestGetEmpty(url, config):
+  return _Request(url, None, config, False, None)
+
+
+def _RequestPost(url, request, config, content_type):
+  return _Request(url, request, config, True, content_type)
+
+
+def _RequestGet(url, request, config):
+  return _Request(url, request, config, False)
```
Comment (TaranFurnell) [inline src/harness/request_handler.py]: _RequestPost and _RequestGet aren't used by anything, correct? Pretty sure the only functions the should be called directly for GET and POST should be:
- RequestPostEmpty (Not sure this is the best name, but naming is hard)
- RequestPostJson
- RequestGet (No need to change it to RequestGetEmpty)

**Comment by janetliuboston** [comment]
Updated. Please review again. Thank you, Taran!

**Comment by janetliuboston** [comment]
Thank you for the review. I'm going to close this for now, as I don't want to disrupt current code with a last minute change like this.

**Pull request closed without merging**